src/coreclr/System.Private.CoreLib/src/System/GC.CoreCLR.cs

```diff
@@ -790,7 +790,41 @@ internal enum GCConfigurationType
         [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "GCInterface_EnumerateConfigurationValues")]
         internal static unsafe partial void _EnumerateConfigurationValues(void* configurationDictionary, delegate* unmanaged<void*, void*, void*, GCConfigurationType, long, void> callback);
 
-        private static int _RefreshMemoryLimit()
+        /// <summary>
+        ///
+        /// Instructs the Garbage Collector to reconfigure itself by detecting the various memory limits on the system.
+        ///
+        /// In addition to actual physical memory limit and container limit settings, these configuration settings can be overwritten:
+        ///
+        /// - GCHeapHardLimit
+        /// - GCHeapHardLimitPercent
+        /// - GCHeapHardLimitSOH
+        /// - GCHeapHardLimitLOH
+        /// - GCHeapHardLimitPOH
+        /// - GCHeapHardLimitSOHPercent
+        /// - GCHeapHardLimitLOHPercent
+        /// - GCHeapHardLimitPOHPercent
+        ///
+        /// Instead of updating the environment variable (which will not be read), these are overridden setting a ulong value in the AppContext.
+        ///
+        /// For example, you can use AppContext.SetData("GCHeapHardLimit", (ulong) 100 * 1024 * 1024) to override the GCHeapHardLimit to a 100M.
+        ///
+        /// As a limitation, for 32-bit systems (e.g. x86, ARM) or OSX, if we do not already have a heap hard limit set, we will not establish a new heap hard limit.
+        ///
+        /// As of now, this API is feature preview only and subject to changes as necessary.
+        ///
+        /// </summary>
+        ///
+        /// <returns> A status code to indicate the operation's result:
```

Comment:
I doubt that "int" return value is going to fly well as a public API design.

Reply:
How about an enum? Wouldn't it be a breaking change if I need to add more error status codes?

Reply:
There is no significant difference. Both of them fall under "Increasing the range of returned values for a property, field, return or out value" case from https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/docs/coding-guidelines/breaking-change-rules.md#property-field-parameter-and-return-values